src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/Validation/ValidationStateEntry.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// An entry in a <see cref="ValidationStateDictionary"/>. Records state information to override the default
+    /// behavior of validation for an object.
+    /// </summary>
+    public class ValidationStateEntry
+    {
+        /// <summary>
+        /// Gets or sets the model prefix associated with the entry.
+        /// </summary>
+        public string Key { get; set; }
```

Comment:
See the need to modify an existing entry in `CompositeModelBinder`. But, pity this `class` can't be immutable, like `ValidationEntry`.